sys/include/net/gcoap.h

```diff
@@ -29,6 +29,14 @@
  * provides functions to generate and send an observe notification that are
  * similar to the functions to send a client request.
  *
+ * *Contents*
+ *
+ * - Server Operation
+ * - Client Operation
+ * - Observe Server Operation
+ * - Implementation Notes
+ * - Implementation Status
```

Comment:
I dug deeper, @miri64, but cannot get the [TOC] tag to work in the context of code documentation. My understanding is that this tag works only in _page_ context [1, see Warning]. A markdown file like mainpage.md is in page context, but code documentation like gcoap.h is not [2].

[1] http://www.stack.nl/~dimitri/doxygen/manual/commands.html#cmdtableofcontents
[2] http://www.stack.nl/~dimitri/doxygen/manual/markdown.html#md_page_header